### ISSUE TITLE
Xvic filename flags for unexpanded programs

### ISSUE THREAD

**Issue by sionisaac** [libretro/vice-libretro#186]
Hi i'm trying to setup retroarch to load all my Vic20 games.  To hopefully avoid buying a C64 maxi.  From researching filename flags in vice I managed to work out that (3k), (8k), (16k) flags work great for the expanded memory games but when it comes to unexpanded memory games depending on the last settings for xvic the games do not run in most cases.  I really don't want to have to setup individual configs for every unexpanded game.  Is there a filename flag that will tell the core to boot with unexpanded ram?

**Comment by sonninnos** [comment]
Based on this comment in the source:
```
/* This function parses the mem config string given as `-memory' and returns
 * the appropriate values or'ed together.
 *
 * basically we accept a comma separated list of options like the following:
 * ""   - no extension
 * none - same
 * all  - all blocks
 * 3k   - 3k space in block 0 (=3k extension cartridge)
 * 8k   - 1st 8k extension block
 * 16k  - 1st and 2nd 8 extension (= 16k extension cartridge)
 * 24k  - 1-3rd extension block (=8k and 16k extension cartridges)
 *
 * 0,1,2,3,5      - memory in respective block
 * 04,20,40,60,a0 - memory at respective address (same as block #s)
 *
 * example: xvic -memory ""
 *
 *            enables unexpanded computer
 *
 *          xvic -memory 60,a0
 *
 *            enables memory in blocks 3 and 5, which is the usual
 *            configuration for 16k rom modules
 *
 * 12/27/96 Alexander Lehmann <alex@mathematik.th-darmstadt.de>
 * Edited by Ettore to fit in the new command-line parsing.
 */
```
Shouldn't then `xvic -memory ""` or `xvic -memory none` do the trick? Or does the core option override that or what is the case exactly?

**Comment by sionisaac** [comment]
The core options are overriding everything.  If I select an 8 or 16k game from the retroarch playlist they play fine because the filename flags seem to be telling the core which memory setting to use.  But if I select a game that doesn't need expanded ram it will mostly not work because the last settings used were for expanded ram.  

So what you are saying is if I want to have this setup so I never have to manually change the memory options to run a game.  I need to setup a command file for every game telling the core to use no expanded ram.  Where as the games requiring expanded ram just need a filename flag to tell the core how to treat it e.g. :-
Alien Demon (8k).org
I know the Vic is a pain in the arse with RAM.  I used to own one.  All I want is to be able to select any game from my playlist and run it without having to bigger about with the core options for expanded memory.  Any way to do this without a CMD file for absolutely every rom?

**Comment by sonninnos** [comment]
I don't quite understand. There is nothing in this core that searches for a "(8k)" in the filename, and thus I can't get it to do anything..

That is why I thought you used that VICE internal command line option.

**Comment by sionisaac** [comment]
I think I read about the filename flags in the Vice online manual.  I tried some games that I'd renamed following the instructions on winvice xvic and they all seemed to work as they should without having to tell the emulator to use specific expanded memory.  So I tried the exact same games on the xvic retroarch core and they seemed to work fine too.  Example:

Run a 16k game from retroarch startup without changing core options.  Close the game and open an 8k game still not changing the options then a 16k game.  But then if I try to run an unexpanded memory game I get a screen full of garbage until I change the core options to unexpanded memory and reset the game.  I think the easiest way for me to proceed is to write a script that will create a command file for every game with the exact memory requirements.  That way my filenames are untouched.  I decided to bite the bullet and order a C64 maxi.  I've read that there are also filename flags for Vic20 progs for that too but they are different to the vice flags.  Oh well it's a learning curve.  Cheers for the help I'll probably modify the command you already said for my script.
